### PR TITLE
Omero tree fixes

### PR DESCRIPTION
**Pull request by will-moore**
Fixes a couple of issues from #9

 - No filtering by default group or user
 - Alphabetical ordering
 - Also caches `childCount` so we don't have to request this from the server multiple times.

To test:

```
$ omero login
# idr.openmicroscopy.org, public, public
$ napari_omero
```

The biggest issue now is that we try to load the whole depth of the tree up-front, instead of waiting for the user to expand each node.
This means that with IDR, you are loading *forever* and never reach the whole tree. But the top of the tree loads OK:

Screenshot 2020-06-25 at 16 26 30

### CODE REVIEW

**Comment by tlambert03** [comment]
oh awesome.  thank you!

> The biggest issue now is that we try to load the whole depth of the tree up-front, instead of waiting for the user to expand each node.

Yeah that was a short-term hack.  I'll follow up and fix the `_populate_tree` method to be lazy again, using `fetchMore` and `canFetchMore` (which you may have seen were commented out for now).  I did want to ask though, I feel like expanding a node (fetching children for just that node) is still much slower than it is in the webclient.  Any thoughts there?  Is there any prefetching going on?

**Comment by tlambert03** [comment]
I've also left out screens for now... since I have no experience with them

**Comment by will-moore** [comment]
Using those `getObjects()` queries might be a bit faster. I noticed that the listChildren() calls are also loading annotations.
Also the caching of childCount should help a little. Can't think of anything else just now.

File: src/napari_omero/widgets/tree_model.py [hunk 1]
```diff
@@ -58,7 +61,9 @@ def _populate_tree(self):
 
         root = self.invisibleRootItem()
         projects = []
-        for project in list(self.gateway.conn.listProjects()):
+
+        for project in list(self.gateway.conn.getObjects(
+                            "Project", opts={'order_by': 'obj.name'})):
```
Comment (tlambert03) [inline src/napari_omero/widgets/tree_model.py:66]: ❤️

File: src/napari_omero/widgets/tree_model.py [hunk 2]
```diff
@@ -34,7 +35,9 @@ def hasChildren(self):
         return bool(self.wrapper.CHILD_WRAPPER_CLASS)
 
     def numChildren(self) -> int:
-        return self.wrapper.countChildren()
+        if self._childCount is None:
+            self._childCount = self.wrapper.countChildren()
+        return self._childCount
```
Comment (tlambert03) [inline src/napari_omero/widgets/tree_model.py:40]: to fix the mypy error, you can either this above:
```
self._childCount: Optional[int] = None
```
or delete `self._childCount = None` and change this to 
```
        if not hasattr(self, '_childCount'):
            self._childCount: int = self.wrapper.countChildren()
        return self._childCount
```

Comment (tlambert03) [reply src/napari_omero/widgets/tree_model.py:40]: working on this right now, adding lazy loading.  much better.  thanks for the tip on getObjects

Comment (tlambert03) [reply src/napari_omero/widgets/tree_model.py:40]: I'm gonna push the merged changes ... so pull if you're also working on it

Comment (tlambert03) [reply src/napari_omero/widgets/tree_model.py]: nevermind... looks like I can't push to your branch.  I'll just merge this one

**Comment by will-moore** [comment]
Working well. Thanks 👍

**Comment by tlambert03** [comment]
there's enough goodness here to merge.  we can address further improvements to the tree later

**Pull request merged**